lib/algos/list/delimiterMaster.js

```diff
@@ -95,12 +95,29 @@ class DelimiterMaster extends Delimiter {
              *   NextMarker to the common prefix instead of the whole key
              *   value. (TODO: remove this test once ZENKO-1048 is fixed)
              *   */
-            if (key === this.prvKey || key === this[this.nextContinueMarker] ||
-                (this.delimiter &&
-                key.startsWith(this[this.nextContinueMarker]))) {
+            if (key === this.prvKey || key === this[this.nextContinueMarker]) {
                 /* master version already filtered */
                 return FILTER_SKIP;
             }
+
+            if (this.delimiter && key.startsWith(this[this.nextContinueMarker])) {
+                // If this[continueMarker] ends with the delimiter then we have to treat it differently.
+                const candidate = key.split(this.delimiter);
+                const len = candidate.length;
+                if (len === 1) {
+                    // case key=foo: this is a version of a previously seen master key.
+                    return FILTER_SKIP;
+                }
+                if (len > 1) {
+                    // key=foo/ or key=foo/bar
+                    if (key[len - 1] === '') {
+                        // key=foo/ : version of a previously seen master key.
+                        return FILTER_SKIP;
+                    }
+                    // Else key=foo/bar (or further nesting as foo/bar/baz/..)
+                    // This case should not be skipped and be handled in this.addContents below.
```

Comment:
Why so? If the key is `foo/bar` and the value of `this[this.nextContinueMarker]` is `foo/`, we must skip that key as it is part of a prefix already seen, I think?

Reply:
this is what gets handled on line 98

Reply:
Can you be more specific

Reply:
I worded this poorly. Should be "should not be skipped *yet*.. ". It does indeed get skipped but this case is handled later in the function. Updated now.